pypesto/collections/collections.py

```diff
@@ -0,0 +1,497 @@
+
+            timepoints.append(np.concatenate(
+                [cond.timepoints for cond in result.conditions], axis=0))
+
+        # stack results in third dimension
+        if output is not None:
+            output = np.stack(output, axis=2)
+        if output_sensi is not None:
+            output_sensi = np.stack(output_sensi, axis=3)
+
+        # formulate as dict
+        self.prediction_arrays = {
+            OUTPUT: output,
+            OUTPUT_SENSI: output_sensi,
+            TIMEPOINTS: np.stack(timepoints, axis=-1)
+        }
+
+    def compute_summary(self,
+                        percentiles_list: Sequence[int] = (5, 20, 80, 95)
+                        ) -> Dict:
+        """
+        This function computes the mean, the median, the standard deviation
+        and possibly percentiles from the collection prediction results.
+        Those summary results are added as a data member to the
+        CollectionPrediction object.
+
+        Parameters
+        ----------
+        percentiles_list:
+            List or tuple of percent numbers for the percentiles
+
+        Returns
+        -------
+        summary:
+            dictionary of predictions results with the keys mean, std, median,
+            percentiles, ...
+        """
+        # check if prediction results are available
+        if not self.prediction_results:
+            raise ArithmeticError('Cannot compute summary statistics from '
+                                  'empty prediction results.')
+        n_conditions = len(self.prediction_results[0].conditions)
+
+        def _stack_outputs(ic: int):
+            """
+            This function groups outputs for different parameter vectors of
+            one ensemble together, if they belong to the same simulation
+            condition, and stacks them in one array
+            """
+            # Were outputs computed
+            if self.prediction_results[0].conditions[ic].output is None:
+                return None
+            # stack predictions
+            output_list = [prediction.conditions[ic].output
+                           for prediction in self.prediction_results]
+            # stack into one numpy array
+            return np.stack(output_list, axis=-1)
+
+        def _stack_outputs_sensi(ic: int):
+            """
+            This function groups output sensitivities for different parameter
+            vectors of one ensemble together, if the belong to the same
+            simulation condition, and stacks them in one array
+            """
+            # Were output sensitivitiess computed
+            if self.prediction_results[0].conditions[ic].output_sensi is None:
+                return None
+            # stack predictions
+            output_sensi_list = [prediction.conditions[ic].output_sensi
+                                 for prediction in self.prediction_results]
+            # stack into one numpy array
+            return np.stack(output_sensi_list, axis=-1)
```

Comment:
Looks rather similar to `condense_to_arrays`. Can this be combined?

Reply:
hm... maybe... They are similar, indeed, but there is one important difference: the `_stack_...` functions keep the substructure of having conditions, and therefore allow for different observables per condition. `condense_to_arrays` doesn't do so, it's the brute force approach crunching everything together. Therefore, it's really only usable if all conditions have the same shape. Will document this a bit more...
I added those functions, because I'm pretty sure they will be helpful soon in one or the other way. condense_to_arrays does more than only stacking, however... Not quite, would rather like to wait until the first person really uses them. Then it's still minor effort to change and it will be easier to see whether or not these functions need to be adapted...